key-script

```diff
@@ -46,20 +41,24 @@ fi
 PW="$($cryptkeyscript "$WELCOME_TEXT")"
 
 if [ "$HASH" = "1" ]; then
-	PW=$(printf %s "$PW" | sha256sum | awk '{print $1}')
+    P2=$(printf %s "$PW" | sha256sum | awk '{print $1}')
+else
+    P2="$PW"
 fi
 
-if check_yubikey_present; then
-	message "Accessing yubikey..."
-    R="$(ykchalresp -2 "$PW" 2>/dev/null || true)"
-	message "Retrieved the response from the Yubikey"
-	if [ "$CONCATENATE" = "1" ]; then
-		printf '%s' "$PW$R"
-	else
-	        printf '%s' "$R"
-	fi
+message "Accessing yubikey (please press the button if required)..."
+R="$(ykchalresp -2 "$P2" 2>/dev/null || true)"
+# If the Yubikey not available or timed out waiting for a button press
```

Comment:
It the comment correct? If a yubikey is not present I would like to boot with an emergency passphrase from a passphrase slot.

Reply:
No this comment applies to the else, I'll update it to make more sense.

In the case where a Yubikey isn't present the entered passphrase will be used, see the else condition.